pymatviz/ptable.py

```diff
@@ -910,18 +916,211 @@ def ptable_hists(
+
+    # Turn off axis of subplots on the grid that don't correspond to elements
+    ax: plt.Axes
+    for ax in axes.flat:
+        ax.axis("off")
+
+    symbol_kwargs = symbol_kwargs or {}
+    for Z in range(1, 119):
+        element = Element.from_Z(Z)
+        symbol = element.symbol
+        row, group = df_ptable.loc[symbol, ["row", "column"]]
+
+        ax = axes[row - 1][group - 1]
+        symbol_kwargs.setdefault("fontsize", 10)
+        ax.text(
+            *symbol_pos,
+            symbol_text(element)
+            if callable(symbol_text)
+            else symbol_text.format(elem=element),
+            ha="center",
+            va="center",
+            transform=ax.transAxes,
+            **symbol_kwargs,
+        )
+        ax.axis("on")  # re-enable axes of elements that exist
+
+        scatter_data = data.get(symbol, [])
+
+        if anno_kwds:
+            defaults = dict(
+                text=lambda hist_vals: si_fmt_int(len(hist_vals)),
+                xy=(0.8, 0.8),
+                xycoords="axes fraction",
+                fontsize=8,
+                horizontalalignment="center",
+                verticalalignment="center",
+            )
+            if callable(anno_kwds):
+                annotation = anno_kwds(scatter_data)
+            else:
+                annotation = anno_kwds
+                anno_text = anno_kwds.get("text")
+                if isinstance(anno_text, dict):
+                    anno_text = anno_text.get(symbol)
+                elif callable(anno_text):
+                    anno_text = anno_text(scatter_data)
+                annotation["text"] = anno_text
+            ax.annotate(**(defaults | annotation))
+
+        if scatter_data:
+            if callable(scatter_kwds):
+                scatter_kwargs = scatter_kwds(scatter_data)
+            else:
+                scatter_kwargs = scatter_kwds
+
+            # Plot without colormap when data len is 2
+            if len(scatter_data) == 2:
+                ax.scatter(scatter_data[0], scatter_data[1], **(scatter_kwargs or {}))
+
+            # Plot with colormap when data len is 3
+            elif len(scatter_data) == 3:
+                ax.scatter(
+                    scatter_data[0],
+                    scatter_data[1],
+                    c=scatter_data[2],
+                    cmap=cmap,
+                    **(scatter_kwargs or {}),
+                )
+
+            ax.tick_params(labelsize=8, direction="in")
+
+        # Disable ticks for elements without data
+        else:
+            ax.set_xticks([])
+
+        # Disable y ticks for all elements
+        ax.set_yticks([])
+
+        # Hide y ticks and right/top boarders
+        for side in ("right", "top"):
+            ax.spines[side].set_visible(b=False)
+        ax.tick_params(axis="y", which="both", length=0)
+
+    # Add colorbar if data dimensionality is 3
+    if isinstance(cmap, Colormap) and {len(data) for data in data.values()} == {3}:
```

Comment:
Using other data types other than `dict` would certainly break this function. I would get it right tomorrow.